### DISCUSSION TITLE
Why can't I stop on this breakpoint in a BeforeSend callback?

### DISCUSSION THREAD

**Discussion by fpw23** [getsentry/sentry-dotnet#2110] [Q&A] [answered]
I am trying to a prefix to all sentry transaction names using the BeforeSend callback but it is not working.  I am doing something similar in the javascript sdk and it works fine.  Am I doing something wrong?  

![image](https://user-images.githubusercontent.com/6162217/211835216-d243c6dd-6fb0-4ac9-bc15-6ad9d63fb0cd.png)

In my code I can always hit the first breakpoint be never the second one.

**Comment by mattjohnsonpint** [comment, marked as answer]
Only error events will call `BeforeSend`.  You would need `BeforeSendTransaction`, which we have on our backlog with #2077.

You could try using a transaction processor instead, such as:

```csharp
public class PrefixTransactionNameProcessor : ISentryTransactionProcessor
{
    private readonly string _namePrefix;

    public PrefixTransactionNameProcessor(string namePrefix)
    {
        _namePrefix = namePrefix;
    }

    public Transaction Process(Transaction transaction)
    {
        ((IEventLike)transaction).TransactionName = $"{_namePrefix} - {transaction.Name}";
        return transaction;
    }
}
```

You'd add it to the options with:

```csharp
o.AddTransactionProcessor(new PrefixTransactionNameProcessor(namePrefix));
```
